lib/dry/system/hanami/resolver.rb

```diff
@@ -1,34 +1,43 @@
+# frozen_string_literal: true
+
 module Dry
   module System
     module Hanami
       module Resolver
         PROJECT_NAME = ::Hanami::Environment.new.project_name
+        PROJECT_FOLDER = "lib/#{PROJECT_NAME}/".freeze
 
         def register_folder!(folder, resolver: ->(k) { k.new })
           all_files_in_folder(folder).each do |file|
-            register_name = file.sub("#{PROJECT_NAME}/", '').gsub('/', '.').gsub(/_repository\z/, '')
+            register_name = file.sub(PROJECT_FOLDER, '').tr('/', '.').sub(/_repository\z/, '')
             register(register_name, memoize: true) { load! file, resolver: resolver }
           end
         end
 
         def all_files_in_folder(folder)
           Dir
-            .entries("#{::Hanami.root}/lib/#{folder}")
-            .select {|f| !File.directory? f}
-            .map! { |file_name| "#{folder}/#{file_name.sub!('.rb', '')}" }
+            .glob("lib/#{folder}/**/*.rb")
+            .map! { |file_name| file_name.sub('.rb', '').to_s }
         end
 
         def load!(path, resolver: ->(k) { k.new })
           load_file!(path)
 
-          unnecessary_part = path[/repositories/] ? "#{PROJECT_NAME}/repositories" : "#{PROJECT_NAME}/"
+          unnecessary_part = case path
+                             when /repositories/
+                               "#{PROJECT_FOLDER}repositories"
+                             when /entities/
+                               "#{PROJECT_FOLDER}entities"
+                             else
+                               PROJECT_FOLDER
+                             end
           right_path = path.sub(unnecessary_part, '')
 
-          resolver.call(Object.const_get(Inflecto.classify(right_path).sub('Statu', 'Status')))
+          resolver.call(Object.const_get(Inflecto.camelize(right_path)))
```

Comment:
Nice trick with `camelize`, I remember, `Inflecto` has problems with `Status` word for classify:
![screenshot 2018-03-31 17 33 28](https://user-images.githubusercontent.com/1147484/38164214-a4fd8cf8-3509-11e8-93b5-0eb3446e1a05.jpeg)

Reply:
Thanks, so far is working on my project great. Should we merge it and open another PR for adding tests?